types/client.d.ts

```diff
@@ -4,211 +4,171 @@ import { Duplex, Readable, Writable } from 'stream'
-		/** Invoked when an error has occurred. */
-		onError?(err: Error): void;
-	}
-}
\ No newline at end of file
+  export interface Options {
+    /** an IPC endpoint, either Unix domain socket or Windows named pipe. Default: `null`. */
+    socketPath?: string | null;
+    /** the timeout after which a socket without active requests will time out. Monitors time between activity on a connected socket. This value may be overriden by *keep-alive* hints from the server. Default: `4e3` milliseconds (4s). */
+    keepAliveTimeout?: number;
+    /** the maximum allowed `idleTimeout` when overriden by *keep-alive* hints from the server. Default: `600e3` milliseconds (10min). */
+    keepAliveMaxTimeout?: number;
+    /** A number subtracted from server *keep-alive* hints when overriding `idleTimeout` to account for timing inaccuries caused by e.g. transport latency. Default: `1e3` milliseconds (1s). */
+    keepAliveTimeoutThreshold?: number;
+    /** The amount of concurrent requests to be sent over the single TCP/TLS connection according to [RFC7230](https://tools.ietf.org/html/rfc7230#section-6.3.2). Default: `1`. */
+    pipelining?: number;
+    /** An options object which in the case of `https` will be passed to [`tls.connect`](https://nodejs.org/api/tls.html#tls_tls_connect_options_callback). Default: `null`. */
+    tls?: TlsOptions | null;
+    /** The maximum length of request headers in bytes. Default: `16384` (16KiB). */
+    maxHeaderSize?: number;
+    /** The amount of time the parser will wait to receive the complete HTTP headers (Node 14 and above only). Default: `30e3` milliseconds (30s). */
+    headersTimeout?: number;
+  }
+
+  export interface DispatchOptions {
+    path: string;
+    method: string;
+    /** Default: `null` */
+    body?: string | Buffer | Uint8Array | Readable | null;
+    /** Default: `null` */
+    headers?: IncomingHttpHeaders | null;
+    /** The timeout after which a request will time out, in milliseconds. Monitors time between receiving a complete headers. Use 0 to disable it entirely. Default: `30e3` milliseconds (30s). */
+    headersTimeout?: number;
+    /** The timeout after which a request will time out, in milliseconds. Monitors time between receiving a body data. Use 0 to disable it entirely. Default: `30e3` milliseconds (30s). */
+    bodyTimeout?: number;
+    /** Whether the requests can be safely retried or not. If `false` the request won't be sent until all preceeding requests in the pipeline has completed. Default: `true` if `method` is `HEAD` or `GET`. */
+    idempotent?: boolean;
+  }
+
+  export interface RequestOptions extends DispatchOptions {
+    opaque?: unknown;
+    /** Default: `null` */
+    signal?: AbortSignal | EventEmitter | null;
+  }
+
+  export interface PipelineOptions extends RequestOptions {
+    /** `true` if the `handler` will return an object stream. Default: `false` */
+    objectMode?: boolean;
+  }
+
+  export interface UpgradeOptions extends DispatchOptions {
```

Comment:
Per the test errors, previously `method` is optional for `upgrade`. But now that its extending from `DispatchOptions` it has become required. 

You can specify `method?: string` here to make it optional again